doc/developer/guide-changes.md

```diff
@@ -0,0 +1,699 @@
+# Developer guide: submitting and reviewing changes
+
+As an engineer at Materialize, the bulk of your day-to-day activities will
+revolve around submitting and reviewing code. This guide documents our code
+review practices and expectations, both for when you are reviewing code and when
+you are receiving a code review.
+
+Much of this guide derives from Google's [Code Review Developer Guide][google-guide],
```

Comment:
Would be nice to call out the places we intentionally deviate strongly from the google guide, if there are any

Reply:
Just one minor one. Added a section on this point.